MatrixSDK/Data/Store/MXFileStore/MXFileStore.m

```diff
@@ -1201,6 +1201,54 @@ - (void)loadUsers
     NSLog(@"[MXFileStore] Loaded %tu MXUsers in %.0fms", users.count, [[NSDate date] timeIntervalSinceDate:startDate] * 1000);
 }
 
+- (NSArray<MXUser *> *)loadUsersWithUserIds:(NSArray<NSString *> *)userIds
+{
+    // Determine which groups to load based on userIds
+    NSMutableDictionary<NSString *, NSMutableArray<NSString *> *> *groups = [NSMutableDictionary dictionary];
+    for (NSString *userId in userIds)
+    {
+        NSString *groupId = [NSString stringWithFormat:@"%tu", userId.hash % 100];
+        
+        NSMutableArray *groupUserdIds = groups[groupId];
+        if (groupUserdIds)
+            [groupUserdIds addObject:userId];
+        else
+            groups[groupId] = [NSMutableArray arrayWithObject:userId];
+    }
+    
+    NSString *usersFolder = [storePath stringByAppendingPathComponent:kMXFileStoreUsersFolder];
+    
+    NSMutableArray<MXUser *> *loadedUsers = [NSMutableArray array];
+    for (NSString *group in groups.allKeys)
+    {
+        @autoreleasepool
+        {
+            NSString *groupFile = [usersFolder stringByAppendingPathComponent:group];
+            
+            // Load stored users in this group
+            @try
+            {
+                NSMutableDictionary <NSString *, MXUser *> *groupUsers = [NSKeyedUnarchiver unarchiveObjectWithFile:groupFile];
+                if (groupUsers)
+                {
+                    NSSet *usersToLoad = [NSSet setWithArray:groups[group]];
+                    for (MXUser *user in groupUsers.allValues)
+                    {
+                        if ([usersToLoad containsObject:user.userId])
+                            [loadedUsers addObject:user];
+                    }
+                }
+            }
+            @catch (NSException *exception)
+            {
+                NSLog(@"[MXFileStore] Warning: MXFileRoomStore file for users group %@ has been corrupted", group);
+            }
+        }
+    }
+    
+    return [loadedUsers copy];
```

Comment:
Why do you return a copy?

Reply:
@manuroe  `loadedUsers` is a mutable object and we can send mutable messages even if variable points to immutable object. That is why I return immutable object